src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -353,14 +353,7 @@ internal virtual void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttribu
             if (lazyCustomAttributesBag.SetAttributes(boundAttributes))
             {
                 this.RecordPresenceOfBadAttributes(boundAttributes);
-                if (addToDiagnostics == null)
-                {
-                    this.AddDeclarationDiagnostics(diagnostics);
-                }
-                else
-                {
-                    addToDiagnostics.AddRange(diagnostics);
-                }
+                declarationDiagnostics.AddRange(diagnostics);
```

Comment:
If we are loading attributes on a local function parameter then that would add the diagnostics to the declaration diagnostics, which this PR is trying to avoid.